packages/images-plugin/lib/plugin.js

```diff
@@ -37,21 +51,54 @@ export default class ImageElementPlugin extends BasePlugin {
     },
   });
 
-  async compareImages(next, driver, ...args) {
-    return await compareImages(...args);
+  /**
+   *
+   * @param {Function} next - The function to be executed after this one
+   * @param {ExtDriver} driver - The driver instance
+   * @param {string} mode
+   * @param {string} firstImage
+   * @param {string} secondImage
+   * @param {Object} [options={}]
+   * @returns {Promise<any>}
+   */
```

Comment:
since we actually expect arguments, the type of this would be something like:

```ts
PluginCommand<[string, string, string, object|undefined]>
```

I can't recall if we can create "named" tuple types, but if we can:

```ts
PluginCommand<[mode: string, firstImage: string, secondImage: string, options: object|undefined]>
```

also, `options` is probably not just any old object, and should eventually be typed.

Reply:
@boneskull I see that `return convertVisualizationToBase64(result);` in `compareImages` function inside compare.js returns `OccurrenceResult`. Should I use `@returns {Promise<OccurrenceResult>}

Reply:
yes, if the function is `async`, the return value must be `Promise<something>`

Reply:
@boneskull  hmmm I see now. Should I use `@type {PluginCommand}` for function that takes`next` and `driver` as the first two arguments as an argument in plugin.js of those plugins?

Reply:
you don't _have_ to, as long as `next` and `driver` are the same type as in `PluginCommand`.